src/library-authoring/component-info/ComponentInfo.tsx

```diff
@@ -147,41 +155,56 @@ const ComponentInfo = () => {
-            variant="outline-primary"
-            className="m-1 text-nowrap flex-grow-1"
-          >
-            {intl.formatMessage(messages.editComponentButtonTitle)}
-          </Button>
-          <Button disabled={publishComponent.isLoading || !canPublish} onClick={publish} variant="outline-primary" className="m-1 text-nowrap flex-grow-1">
-            {intl.formatMessage(messages.publishComponentButtonTitle)}
-          </Button>
-          <ComponentMenu usageKey={usageKey} />
-        </div>
-      )}
-      <AddComponentWidget />
-      <Tabs
-        variant="tabs"
-        className="my-3 d-flex justify-content-around"
-        defaultActiveKey={COMPONENT_INFO_TABS.Preview}
-        activeKey={tab}
-        onSelect={setSidebarTab}
-      >
-        <Tab eventKey={COMPONENT_INFO_TABS.Preview} title={intl.formatMessage(messages.previewTabTitle)}>
-          <ComponentPreview />
-        </Tab>
-        <Tab eventKey={COMPONENT_INFO_TABS.Manage} title={intl.formatMessage(messages.manageTabTitle)}>
-          <ComponentManagement />
-        </Tab>
-        <Tab eventKey={COMPONENT_INFO_TABS.Details} title={intl.formatMessage(messages.detailsTabTitle)}>
-          <ComponentDetails />
-        </Tab>
-      </Tabs>
-    </Stack>
+    <>
+      <Stack>
+        {!readOnly && (
+          <div className="d-flex flex-wrap">
+            <Button
+              {...(canEdit ? { onClick: () => openComponentEditor(usageKey) } : { disabled: true })}
+              variant="outline-primary"
+              className="m-1 text-nowrap flex-grow-1"
+            >
+              {intl.formatMessage(messages.editComponentButtonTitle)}
+            </Button>
+            <Button
+              disabled={publishComponent.isLoading || !canPublish}
+              onClick={openPublishConfirmation}
+              variant="outline-primary"
+              className="m-1 text-nowrap flex-grow-1"
+            >
+              {intl.formatMessage(messages.publishComponentButtonTitle)}
+            </Button>
+            <ComponentMenu usageKey={usageKey} />
+          </div>
+        )}
+        <AddComponentWidget />
+        <Tabs
+          variant="tabs"
+          className="my-3 d-flex justify-content-around"
+          defaultActiveKey={COMPONENT_INFO_TABS.Preview}
+          activeKey={tab}
+          onSelect={setSidebarTab}
+        >
+          <Tab eventKey={COMPONENT_INFO_TABS.Preview} title={intl.formatMessage(messages.previewTabTitle)}>
+            <ComponentPreview />
+          </Tab>
+          <Tab eventKey={COMPONENT_INFO_TABS.Manage} title={intl.formatMessage(messages.manageTabTitle)}>
+            <ComponentManagement />
+          </Tab>
+          <Tab eventKey={COMPONENT_INFO_TABS.Details} title={intl.formatMessage(messages.detailsTabTitle)}>
+            <ComponentDetails />
+          </Tab>
+        </Tabs>
+      </Stack>
+      <PublishConfirmationModal
+        isOpen={isOpenPublisConfirmation}
```

Comment:
```suggestion
        isOpen={isOpenPublishConfirmation}
```

Nit. Though I also think `isPublishConfirmationOpen` sounds much better.

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/1677/commits/8953ef75851f0384ebef587fd74f20a393f1416b